tools/README.md

```diff
@@ -0,0 +1,154 @@
+## Vitest config factory and setup presets
+
+This folder contains utilities to centralize and standardize Vitest configuration across the monorepo.
+
+### Files
+
+- `vitest-config-factory.ts`: builds typed Vitest configs with sensible defaults.
+- `vitest-setup-presets.ts`: reusable groups of `setupFiles` paths by test kind.
+
+### Goals
+
+- Reduce duplication across `vitest.*.config.ts` files.
+- Keep per-package intent clear with minimal overrides.
+- Provide safe defaults and easy extension points.
+
+### Quick start
+
+Use the factory in your suite configs (project root is required):
+
+```ts
+/// <reference types="vitest" />
+import { createE2eConfig, createIntConfig, createUnitConfig } from '../../tools/vitest-config-factory.js';
+
+export default createUnitConfig('core', {
+  projectRoot: new URL('../../', import.meta.url),
+});
+```
+
+Creators:
+
+- `createUnitConfig(projectKey, { projectRoot, ...options })`
+- `createIntConfig(projectKey, { projectRoot, ...options })`
+- `createE2eConfig(projectKey, { projectRoot, ...options })`
+
+`projectKey` is used for cache and coverage directories.
+
+### Defaults
+
+Common to all kinds:
+
+- `reporters: ['basic']`, `globals: true`, `environment: 'node'`
+- `alias: tsconfigPathAliases()`
+- `pool: 'threads'` with `singleThread: true`
+- Cache directories resolved from `projectRoot` (absolute paths)
+
+Coverage:
+
+- Unit/Int: enabled by default, reports to `<projectRoot>/coverage/<project>/<kind>-tests`
+- E2E: disabled by default
+- Default exclude: `['mocks/**', '**/types.ts']`
+
+Global setup:
+
+- Unit/Int: `['<projectRoot>/global-setup.ts']` by default
+- E2E: none by default (set per-suite if needed)
+
+Include patterns:
+
+- Unit: `src/**/*.unit.test.*`
+- Int: `src/**/*.int.test.*`
+- E2E: `tests/**/*.e2e.test.*`
```

Comment:
This information is tightly coupled to our implementation. MAybe link the perset directly here instead of repeating?